### ISSUE TITLE
[COMPONENT] header

### ISSUE THREAD

**Issue by kodeaben** [kirbydesign/designsystem#2743]
- [x] I have written a [good issue](https://github.com/kirbydesign/designsystem/wiki/The-Good%3A-Issue) 

## Describe the solution you'd like
The issue is a sub-issue to #2740 

### NB! This issue must be read alongside the guidelines, which is defined in [Zeplin -> Header](https://app.zeplin.io/project/60337f8fa4b49814fda1b48a/screen/6399d4f89a3811140bcc90d5)

![simple-header](https://user-images.githubusercontent.com/114976818/211747470-e5a2e6ba-9d30-4308-ab7f-6a45f7fac331.png)

# Acceptance criteria  (AC): 
The AC is witten in the order they appear in the image above, from top to bottom

- [x] Has an Avatar field of type `Avatar`
- [x] Has an optional status information field  of type `Flag` (3) 
- [x] The header has a Title (h1). (4). Field is required
- [x] The header title has "Font scaling", if the title is more than 2 lines.(4) // see Page / fit heading directive. Field is optional
- [x] The header can have 0-2 subtitles. font-size 14px. It must not linebreak, if it does, truncate (8)
- [x] A custom section (9) An open slot where you can paste anything in. if empty, collaps. We won't make restriction for this field. Field is optional
- [x] Actions. An open slot where you can paste anything in e.g. ButtonGroup. if empty, collaps. We won't make restriction for this field. Field is optional
- [x] If an optional field is empty, collapse it

## Padding
These Acs cannot stand alone. Please also use the Zeplin sketches.

![Screenshot 2023-01-13 at 13 01 27](https://user-images.githubusercontent.com/114976818/212315835-930a44a3-4355-4c63-8c75-e9af7eb578c6.png)

#### This section describes the "outer padding".
- [x] padding left is 16px (additional 16px comes from page)
- [x] Padding-left for button group slot is 8px;
- [x] padding right is 16px (additional 16px comes from page)
- [x] padding top is 4px (the total padding is 16px, the remaining 12 comes from the `Page`
- [x] padding bottom is 32px

#### This section describes the padding between each element.

- [x] Toolbar -> Title : padding 2px;
- [x] Title -> subtitle: padding: 2px;
- [x] Subtitle -> custom section : padding: 8px;
- [x] Custom section -> Button group: padding: 24px;

### Button group
The Button group is out of scope, but the slot must be prepared.
There are no requirements to the button group, as all styling will happen inside the Button group itself.

## Desktop
![Screenshot 2023-01-13 at 14 04 26](https://user-images.githubusercontent.com/114976818/212326564-42da60e7-11b9-44b4-ad1b-a77bf91eb15b.png)

Once the breakpoint for Desktop width is met, the buttongroup should move up next to the h1 styled element(title or value depending on if value is defined), as shown in the image above.

// Question: What if there isn't a button group? Should the rest be centered?
## Have you considered any alternatives?
No

## Are there any additional context?
No

<hr />

## Checklist:

The following tasks should be carried out in sequence in order to follow [the process of contributing](https://github.com/kirbydesign/designsystem/blob/main/.github/CONTRIBUTING.md/#the-process-of-contributing) correctly.

### Refinement

- [x] Request that the issue is [UX refined](https://github.com/kirbydesign/designsystem/blob/main/.github/CONTRIBUTING.md/#ux-refinement); do not proceed until this is done.
- [x] Request that the issue is [tech refined](https://github.com/kirbydesign/designsystem/blob/main/.github/CONTRIBUTING.md/#tech-refinement); do not proceed until this is done.

### Implementation 
The contributor who wants to implement this issue should: 

- [x] Make sure you have read: "[Before you get coding](https://github.com/kirbydesign/designsystem/blob/main/.github/CONTRIBUTING.md/#before-you-get-coding)".
- [x] Signal to others you are working on the issue by assigning yourself.
- [x] Create a branch from the [develop branch](https://github.com/kirbydesign/designsystem/tree/develop) following our [branch naming convention](https://github.com/kirbydesign/designsystem/wiki/The-Good%3A-Branch). 
- [x] Publish a WIP implementation to Github as a draft PR and ask for feedback. 
- [x] Make sure you have implemented tests following the guidelines in: "[The good: Test](https://github.com/kirbydesign/designsystem/wiki/The-Good%3A-Test)".
- [x] Update the [cookbook](https://cookbook.kirby.design) with examples and showcases.

### Review
Once the issue has been implemented and is ready for review:

- [x] Do a [self-review](https://github.com/kirbydesign/designsystem/wiki/The-Good%3A-Self-review). 
- [x] Create a pull-request. If you created a draft PR during implementation you can just mark that as "ready for review".
